systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java

```diff
@@ -122,6 +125,30 @@ void testKafkaTopicRFLowerThanMinInSyncReplicas() {
         assertThat(StatefulSetUtils.ssSnapshot(kafkaName), is(not(kafkaPods)));
     }
 
+    @Test
+    void testKafkaPodCrashLooping() throws InterruptedException {
```

Comment:
Does this actually work? Why should it decide to roll the pod? In your test the configuration is up to date so I would not expect it to roll the pod. I think the test should do the following:

1) Deploy the Kafka cluster with UseParNewGC
2) Wait for it to crash loop
3) Remove UseParNewGC from the config
4) Wait for it to roll
5) Wait for it to be ready

With the test as it is written right now, I would not expect the pods to be rolled at all since there was no change to anything. It should not roll them just because they are crashlooping ... but if it needs to roll them and they are crashlooping it should not wait for a successful connection tot he broker.

Reply:
I see your points. Let me retest it properly.